### PR TITLE
ensures buffer is available when converting into ETH address

### PR DESCRIPTION
**Pull request by koshilife**
### Summary

refs [this forum page](https://community.blockcerts.org/t/v3-verifier-i-got-an-error-buffer-is-not-defined/3226)

>I tried to verify the generated VC on my local verifier and https://www.blockcerts.org/.
>I got an error “Buffer is not defined” like this.

### Other Information

I tested manually and confirmed that it was able to verify processes in my local environment, using [this VC](https://gist.github.com/koshilife/298217a894e47448a5e36168d26560ef).

Screen Shot 2022-06-21 at 20 12 38

### CODE REVIEW

**Comment by koshilife** [comment]
Thank you for your quick review.
Sure, I got it. please give me a time for adding the test.

**Review by lemoustachiste — APPROVED** [review]

File: test/application/inspectors/did/deriveIssuingAddressFromPublicKey.test.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    it('should return the address of Bitcoin Mainnet', function () {
+      const address = deriveIssuingAddressFromPublicKey(publicKey, BLOCKCHAINS.bitcoin);
+      expect(address).toBe('127ZSsk5cWiubyDBkocJdW9dFYLN5N1jHF');
+    });
+  });
+
+  describe('given the argument chain was Mocknet', function () {
+    it('should return the address of Bitcoin Mocknet', function () {
+      const address = deriveIssuingAddressFromPublicKey(publicKey, BLOCKCHAINS.mocknet);
+      expect(address).toBe('127ZSsk5cWiubyDBkocJdW9dFYLN5N1jHF');
```
[the remark this answers is not in the document]
Comment (koshilife) [reply test/application/inspectors/did/deriveIssuingAddressFromPublicKey.test.ts:23]: Thanks.

FYI:
I would try to explain to you the reason why the `Mocknet` address is the same as the `Bitcoin` address.

The network name `mocknet` is not defined in `bitcoinjs-lib`.

<details>
<summary>(details) confirmation codes that mocknet is not defined:</summary>

```sh
$ npm list --depth=0 | grep bitcoinjs-lib 
├── bitcoinjs-lib@5.2.0

$ node
Welcome to Node.js v16.13.0.
Type ".help" for more information.
> const bitcoin = require('bitcoinjs-lib');
> bitcoin.networks['bitcoin']
{
  messagePrefix: '\x18Bitcoin Signed Message:\n',
  bech32: 'bc',
  bip32: { public: 76067358, private: 76066276 },
  pubKeyHash: 0,
  scriptHash: 5,
  wif: 128
}
> bitcoin.networks['mocknet']
undefined
```
</details>

So, in the case of giving `mocknet`,  the convertor method `p2pkh` would be given the network arg is undefined.
like this.

```js
bitcoin.payments.p2pkh({ pubkey: publicKeyBuffer, network: undefined }).address
```

According to [the library code](https://github.com/bitcoinjs/bitcoinjs-lib/blob/7aaef308e0b1c9dddd7ba8049780e59ecb4f6f8c/ts_src/payments/p2pkh.ts#L43), if the network arg is undefined, the bitcoin's network setting would be set as a default.

That's why.

**Comment by koshilife** [comment]
@lemoustachiste Thanks for the merge!
Do you have a plan to release a new npm package and deploy this to [blockcerts.org](https://www.blockcerts.org/)?

**Comment by lemoustachiste** [comment]
it should automatically publish after merge with semantic release, but I see there is a node version mismatch that incapacitates the process. I'm going to fix this. After that it should land in the next few hours on blockcerts.org

**Comment by koshilife** [comment]
Thanks for the explanation.
That's cool. Alright, I got the situation. I appreciate your efforts.

**Pull request merged**